From 49dbe60d4b3c6f111911c8217bc5e7da5a4ba0d0 Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Wed, 31 May 2023 18:50:12 +0200
Subject: [PATCH] pam: add call to pam_umask

Setting umask for user sessions via UMASK setting in /etc/login.defs is
a well-known feature. Let's make sure that user manager also runs with
this umask value.

Follow-up for 5e37d1930b41b24c077ce37c6db0e36c745106c7.

(cherry picked from commit 159f1b78576ce91c3932f4867f07361a530875d3)

Resolves: RHEL-28048
---
 src/login/systemd-user.m4 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/login/systemd-user.m4 b/src/login/systemd-user.m4
index eb291beaed..a194a636d6 100644
--- a/src/login/systemd-user.m4
+++ b/src/login/systemd-user.m4
@@ -10,4 +10,5 @@ session required pam_selinux.so nottys open
 session required pam_loginuid.so
 session optional pam_keyinit.so force revoke
 session required pam_namespace.so
+session optional pam_umask.so silent
 session optional pam_systemd.so
